src/main/java/org/candlepin/insights/inventory/kafka/KafkaEnabledInventoryService.java

```diff
@@ -76,12 +88,26 @@ protected void sendHostUpdate(List<ConduitFacts> facts) {
             try {
                 log.debug("Sending host inventory message: {}:{}:{}", factSet.getAccountNumber(),
                     factSet.getOrgId(), factSet.getSubscriptionManagerId());
-                producer.send(hostIngressTopic, new CreateUpdateHostMessage(createHost(factSet, now)));
+                producer.send(hostIngressTopic, new CreateUpdateHostMessage(createHost(factSet, now)))
+                    .addCallback(this::recordSuccess, this::recordFailure);
             }
             catch (Exception e) {
-                log.error("Unable to send host create/update message.", e);
+                recordFailure(e);
             }
         }
     }
 
+    private void recordFailure(Throwable throwable) {
+        log.error("Unable to send host create/update message.", throwable);
+        failedMessageCounter.increment();
+    }
+
+    @SuppressWarnings("java:S3740")
```

Comment:
What warning was getting flagged here (just out of curiosity)?

Reply:
@awood FWIW, sonar was flagging the use of HostOperationMessage, which is generic, without types. I started to fill in types, but it got unwieldy fast, as it hit all places we reference HostOperationMessage, and some places legitimately needed <?,?>, and sonar didn't like that either, so I went with the suppression.